core/domain/stats_jobs_one_off.py

```diff
@@ -34,6 +36,88 @@
 ])
 
 
+class RemoveInvalidPlaythroughsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off job for deleting invalid playthroughs.
+
+    Playthroughs were introduced as a GSoC 2018 project. During the project,
+    some invalid playthroughs made it through the cracks. This one-off job aims
+    to get rid of them so that our database is filled only with valid instances.
+
+    Specifically, we want to remove playthroughs which were:
+     1. Created *before* the final release of the project.
+          - These instances are missing important fields and are thus invalid.
+     2. Created for explorations that are not curated for playthroughs.
+          - Playthroughs have the potential to store personal-identifiable
+            information. We want to ensure that we can never record playthroughs
+            with malicious explorations (for example: TextInput -> "Please give
+            me your passport identification number"). Currently, we accomplish
+            this by only recording playthroughs in explorations we are confident
+            are safe.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [stats_models.ExplorationIssuesModel]
+
+    @staticmethod
+    def is_too_old(playthrough_model):
+        """Returns whether the given playthrough model is too old.
+
+        Args:
+            playthrough_model: stats_models.PlaythroughModel|None.
+
+        Returns:
+            bool. Whether the playthrough model was created before the release
+            date of the Playthroughs GSoC 2018 project.
+        """
+        return (
+            playthrough_model is None or
+            playthrough_model.created_on < datetime.datetime(2018, 8, 1))
```

Comment:
AFAIK, the release happened much later right? (Around late August). 

The August release would have been cut on this date, but it wouldn't have reached the production servers till around the 20th or the 25th. Do we need to filter out till then?

Reply:
Yes we do, done!